compiler/rustc_typeck/src/check/cast.rs

```diff
@@ -767,7 +767,7 @@ impl<'a, 'tcx> CastCheck<'tcx> {
     ) -> Result<CastKind, CastError> {
         // array-ptr-cast.
 
-        if m_expr.mutbl == hir::Mutability::Not && m_cast.mutbl == hir::Mutability::Not {
+        if m_expr.mutbl == m_cast.mutbl {
```

Comment:
do you still want to adjust this to allow for `&mut [T; M] as *const T`? 

I personally don't mind either way:shrug:

Reply:
Given that we allow mut-ref-to-mut-ptr I thought it would be more consistent to allow this as well. I implemented this and updated an existing test to test this case too.